library/std/src/lazy.rs

```diff
@@ -348,22 +342,12 @@ impl<T> SyncOnceCell<T> {
     /// ```
     #[unstable(feature = "once_cell", issue = "74465")]
     pub fn take(&mut self) -> Option<T> {
-        mem::take(self).into_inner()
-    }
-
-    /// Takes the wrapped value out of a `SyncOnceCell`.
-    /// Afterwards the cell is no longer initialized.
-    ///
-    /// Safety: The cell must now be free'd WITHOUT dropping. No other usages of the cell
-    /// are valid. Only used by `into_inner` and `drop`.
-    unsafe fn take_inner(&mut self) -> Option<T> {
```

Comment:
:+1: I was thinking about removing `take_inner` as well (its contract is just weird), but I didn't know about those useful MaybeUninit helpers!

Reply:
That's because they didn't exist yet. :) I added `assume_init_drop` in https://github.com/rust-lang/rust/pull/76484 just a few days ago.